bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/Region.java

```diff
@@ -616,7 +623,204 @@ public void translate (Point pt) {
 @Override
 public String toString () {
 	if (isDisposed()) return "Region {*DISPOSED*}";
-	return "Region {" + handle + "}";
+	return "Region {" + zoomToHandle.entrySet().stream().map(entry -> entry.getValue() + "(zoom:" + entry.getKey() + ")").collect(Collectors.joining(","));
 }
 
+@FunctionalInterface
+private interface OperationStrategy {
+	void apply(Operation operation, long handle, int zoom);
+}
+
+abstract class Operation {
+	OperationStrategy operationStrategy;
+
+	void apply(long handle, int zoom) {
+		operationStrategy.apply(this, handle, zoom);
+	}
+
+	abstract void add(long handle, int zoom);
+
+	abstract void subtract(long handle, int zoom);
+
+	abstract void intersect(long handle, int zoom);
+
+	abstract void translate(long handle, int zoom);
+}
+
+private class OperationWithRectangle extends Operation {
+
+	Rectangle data;
+
+	OperationWithRectangle(OperationStrategy operationStrategy, Rectangle data) {
+		this.operationStrategy = operationStrategy;
+		this.data = data;
+	}
+
+	@Override
+	void add(long handle, int zoom) {
+		Rectangle bounds = getScaledRectangle(zoom);
+		addInPixels(handle, bounds.x, bounds.y, bounds.width, bounds.height);
+	}
+
+	@Override
+	void subtract(long handle, int zoom) {
+		Rectangle bounds = getScaledRectangle(zoom);
+		subtractInPixels(handle, bounds.x, bounds.y, bounds.width, bounds.height);
+	}
+
+	@Override
+	void intersect(long handle, int zoom) {
+		Rectangle bounds = getScaledRectangle(zoom);
+		intersectInPixels(handle, bounds.x, bounds.y, bounds.width, bounds.height);
+	}
+
+	@Override
+	void translate(long handle, int zoom) {
+		// No implementation required
```

Comment:
I would prefer to throw an `OperationNotSupportedException` (in all the unimplemented methods), so that in case someone starts using this method, they are informed about the missing implementation rather than doing nothing.

Reply:
It is a check exception and it will change the ssignature of all the methods. I already tried it and I also needed to add the throws in all the client calls. Do we not have an alternative?